contentpacks/khanacademy.py

```diff
@@ -429,15 +429,22 @@ def retrieve_exercise_dict(lang=None, force=False) -> str:
 
 @cache_file
 def download_and_clean_kalite_data(url, path, lang="en") -> str:
-    data = requests.get(url)
     attempts = 1
-    while data.status_code != 200 and attempts <= 100:
-        time.sleep(30)
+    while attempts < 100:
         data = requests.get(url)
-        attempts += 1
-
-    if data.status_code != 200:
-        raise requests.RequestException("Got error requesting KA data: {}".format(data.content))
+        try:
+            data.raise_for_status()
+            break
+        except requests.RequestException as e:
+            logging.warning("Attempt {attempt}: Got error while requesting KA data: {e!s:<80}".format(
+                attempt=attempts,
+                e=e)
+            )
+            attempts += 1
```

Comment:
Yeah, it made some of the runs more robust, but now KA is just suddenly cutting off our connection now :( I think we've hit some kind of rate limiter on their side.